pandas/core/indexes/numeric.py

```diff
@@ -120,15 +121,30 @@ def _validate_fill_value(self, value):
             # force conversion to object
             # so we don't lose the bools
             raise TypeError
-        elif isinstance(value, str) or lib.is_complex(value):
-            raise TypeError
         elif is_scalar(value) and isna(value):
             if is_valid_nat_for_dtype(value, self.dtype):
                 value = self._na_value
+                if self.dtype.kind != "f":
```

Comment:
can you add a doc-string for _validate_fill_value as its not clear when its supposed to raise

Reply:
updated + greenish